### DISCUSSION TITLE
WARNING: The directory '/home/pibirdnet/.cache/pip' or its parent directory is not owned or is not writable by the current user.

### DISCUSSION THREAD

**Discussion by srchild** [mcguirepr89/BirdNET-Pi#762] [General]
I'm getting this error showing when I run update. but my permissions/ownerships appear to be correct.

`WARNING: The directory '/home/pibirdnet/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.`

Is it a spurious error message or do I need to adjust some permissions/ownerships?

dir -la /home/pibirdnet
total 11696
drwxr-xr-x 10 pibirdnet pibirdnet    4096 Feb 15 16:40 .
drwxr-xr-x  3 root       root          4096 Apr  4  2022 ..
drwxr-xr-x  3 pibirdnet pibirdnet    4096 Feb 15 12:14 .apprise
-rw-------  1 pibirdnet pibirdnet   14008 Feb 19 16:44 .bash_history
-rw-r--r--  1 pibirdnet pibirdnet     220 Apr  4  2022 .bash_logout
lrwxrwxrwx  1 pibirdnet pibirdnet      44 Feb 15 13:48 .bashrc -> /home/pibirdnet/BirdNET-Pi/templates/bashrc
-rw-rw-r--  1 pibirdnet pibirdnet 7895794 Feb 15 16:23 birddb.csv
drwxrwxr-x 10 pibirdnet pibirdnet    4096 Feb 16 08:38 BirdNET-Pi
-rw-rw-r--  1 pibirdnet pibirdnet 3837952 May 20  2022 birds.db
drwxrwxrwx 14 pibirdnet pibirdnet    4096 Feb  1 00:00 BirdSongs
drwxr-xr-x  4 pibirdnet pibirdnet    4096 May 20  2022 .cache
drwx------  6 pibirdnet pibirdnet    4096 Feb 15 12:15 .config
drwxr-xr-x  3 pibirdnet pibirdnet    4096 Feb 15 12:14 .config.apprise
-rw-rw-r--  1 pibirdnet pibirdnet    1712 Feb  6 15:55 dead.letter
lrwxrwxrwx  1 pibirdnet pibirdnet      43 Feb 15 13:48 .gotty -> /home/pibirdnet/BirdNET-Pi/templates/gotty
-rw-rw-r--  1 pibirdnet pibirdnet  111976 May 20  2022 installation-2022-05-20.txt
-rw-rw-r--  1 pibirdnet pibirdnet   36764 Feb 15 13:55 installation-2023-02-15.txt
-rw-------  1 pibirdnet pibirdnet     901 Feb 15 12:13 .joe_state
-rw-------  1 pibirdnet pibirdnet      78 Feb 15 14:38 .lesshst
drwxrwxr-x 13 pibirdnet pibirdnet    4096 Feb 15 13:49 phpsysinfo
-rw-r--r--  1 pibirdnet pibirdnet     807 Apr  4  2022 .profile
-rw-------  1 pibirdnet pibirdnet    3906 May 30  2022 .sqlite_history
drwx------  2 pibirdnet pibirdnet    4096 Dec 27 18:36 .ssh

dir -la /home/pibirdnet/.cache
total 16
drwxr-xr-x  4 pibirdnet pibirdnet 4096 May 20  2022 .
drwxr-xr-x 10 pibirdnet pibirdnet 4096 Feb 15 16:40 ..
drwxr-xr-x  2 pibirdnet pibirdnet 4096 May 20  2022 matplotlib
drwxr-xr-x  4 pibirdnet pibirdnet 4096 May 20  2022 pip

dir -la /home/pibirdnet/.cache/pip
total 16
drwxr-xr-x  4 pibirdnet pibirdnet 4096 May 20  2022 .
drwxr-xr-x  4 pibirdnet pibirdnet 4096 May 20  2022 ..
drwxr-xr-x 18 pibirdnet pibirdnet 4096 May 20  2022 http
drwxr-xr-x  4 pibirdnet pibirdnet 4096 May 20  2022 wheels

dir -la /home/pibirdnet/.cache/pip/http
total 72
drwxr-xr-x 18 pibirdnet pibirdnet 4096 May 20  2022 .
drwxr-xr-x  4 pibirdnet pibirdnet 4096 May 20  2022 ..
drwxr-xr-x 16 pibirdnet pibirdnet 4096 Feb 15 13:52 0
drwxr-xr-x 17 pibirdnet pibirdnet 4096 Feb 15 13:52 1
drwxr-xr-x 16 pibirdnet pibirdnet 4096 May 20  2022 2
drwxr-xr-x 13 pibirdnet pibirdnet 4096 May 20  2022 3
drwxr-xr-x 14 pibirdnet pibirdnet 4096 May 20  2022 4
drwxr-xr-x 13 pibirdnet pibirdnet 4096 Feb 15 13:50 5
drwxr-xr-x 15 pibirdnet pibirdnet 4096 May 20  2022 6
drwxr-xr-x 16 pibirdnet pibirdnet 4096 Feb 15 13:51 7
drwxr-xr-x 16 pibirdnet pibirdnet 4096 Feb 15 13:52 8
drwxr-xr-x 13 pibirdnet pibirdnet 4096 May 20  2022 9
drwxr-xr-x 16 pibirdnet pibirdnet 4096 Feb 15 13:49 a
drwxr-xr-x 14 pibirdnet pibirdnet 4096 Feb 15 13:51 b
drwxr-xr-x 15 pibirdnet pibirdnet 4096 Feb 15 13:52 c
drwxr-xr-x 13 pibirdnet pibirdnet 4096 May 20  2022 d
drwxr-xr-x 16 pibirdnet pibirdnet 4096 Feb 15 13:52 e
drwxr-xr-x 11 pibirdnet pibirdnet 4096 May 20  2022 f

dir -la /home/pibirdnet/.cache/pip/wheels
total 16
drwxr-xr-x 4 pibirdnet pibirdnet 4096 May 20  2022 .
drwxr-xr-x 4 pibirdnet pibirdnet 4096 May 20  2022 ..
drwxr-xr-x 3 pibirdnet pibirdnet 4096 May 20  2022 65
drwxr-xr-x 3 pibirdnet pibirdnet 4096 May 20  2022 87

**Comment by ehpersonal38** [comment]
Thankfully that's not an error, just a warning. It's because the update script is called by the webserver which is owned by `caddy`. But it doesn't need caching so there's nothing to worry about.

**Reply by srchild** [reply]
I see. Thanks
